assets/js/elife-loader.js

```diff
@@ -1,15 +1,33 @@
 (function (window) {
   'use strict';
 
-  try {
-    var scriptPaths,
-        $body;
-    if (
+  function browserHasMinimumFeatureSupport () {
+    return (
       !!window.localStorage &&
       !!(window.document.createElement('div')).dataset &&
       typeof window.document.querySelector === 'function' &&
       typeof window.addEventListener === 'function'
-    ) {
+    );
+  }
+
+  function isNetworkInformationAvailable(connection) {
+    return !!connection && typeof connection.effectiveType === 'string';
+  }
+
+  function networkIsDefinitelySlow () {
+    if (!isNetworkInformationAvailable(navigator.connection)) {
+      return false;
+    }
+    if (navigator.connection.effectiveType.indexOf('2g') > -1) {
+      return true;
+    }
+    return false;
+  }
+
+  try {
+    var scriptPaths,
+        $body;
+    if (browserHasMinimumFeatureSupport() && !networkIsDefinitelySlow()) {
       scriptPaths = window.elifeConfig.scriptPaths;
```

Comment:
This is what's currently breaking the tests, as `window.elifeConfig` is `undefined` at this point, and trying to reference a property of `undefined` is an error.

Could avoid this by passing the config object as an argument to `init()` (see comment for line 27 re `init`), so instead of lines 28 and 31 within `init`, something like:

```
function init(config) {
  var scriptPaths = config.scriptPaths;
  ...
}
```

and to invoke it:

```
init(window.elifeConfig || {});
```

The parametrised `init` function could be a good candidate for more tests, too, at least in order to verify that `scriptPaths` can be derived correctly when it's available, and that its absence doesn't break things.

Reply:
Completed